src/fetchers/wakatime-fetcher.js

```diff
@@ -4,29 +4,30 @@ import { MissingParamError } from "../common/utils.js";
 /**
  * WakaTime data fetcher.
  *
- * @param {{username: string, api_domain: string, range: string}} props Fetcher props.
+ * @param {{username: string, api_domain: string}} props Fetcher props.
  * @returns {Promise<WakaTimeData>} WakaTime data response.
  */
-const fetchWakatimeStats = async ({ username, api_domain, range }) => {
+const fetchWakatimeStats = async ({ username, api_domain }) => {
   if (!username) throw new MissingParamError(["username"]);
 
-  try {
-    const { data } = await axios.get(
-      `https://${
-        api_domain ? api_domain.replace(/\/$/gi, "") : "wakatime.com"
-      }/api/v1/users/${username}/stats/${
-        range || "all_time"
-      }?is_including_today=true`,
-    );
-
-    return data.data;
-  } catch (err) {
-    if (err.response.status < 200 || err.response.status > 299) {
-      throw new Error(
-        "Wakatime user not found, make sure you have a wakatime profile",
+  // Loop through available ranges to get user data.
+  for (const range of ["all_time", "last_7_days", "last_year"]) {
```

Comment:
Perhaps swap the order of "last_year" and "last_7_days", since if the user sets the preference to option "Last year", we may be accessible to both "last_year" and the narrower range "last_7_days", when in fact the user might want to show the range "last_year". Therefore an order from a wider range to a narrower one may sound better.

Reply:
@hakula139 I performed some testing, and it looks like Waka Time only allows people to access the range that is public. I, however, agree that from a time perspective it makes sense to swap the two 👍🏻.

Reply:
Fixed in [d80d766](https://github.com/anuraghazra/github-readme-stats/pull/2707/commits/d80d7668b6770ca27b637cebd7ae95da407a94f0).